[Feat] 인풋 디자인 시스템 생성

yonghyeun opened the pull request:
# 관련 이슈 번호
#close 18
# 설명

![image](https://github.com/user-attachments/assets/a622044e-f33d-4f97-8aee-1ebf5b188bf8)

# 첨부 파일 (Optional)

# 레퍼런스 (Optional)

yonghyeun committed:
feat[#18] : 디자인 시스템 : BaseInput 컴포넌트 생성

- 해당 컴포넌트는 인풋 디자인 시스템의 대부분 컴포넌트의 근간이 되는 컴포넌트
- 높은 추상화수준을 가져 props 를 받아 다양한 디자인 시스템을 생성

yonghyeun committed:
feat[#18] : Storybook meta title 컨벤션에 따라 수정

yonghyeun committed:
feat[#18] : SearchTextInput 컴포넌트 생성

yonghyeun committed:
feat[#18] : OutLInedTextInput 컴포넌트 생성

yonghyeun committed:
feat[#18] : BaseInput 컴포넌트 컴포넌트 타입 수정

yonghyeun committed:
feat[#18] : PlainTextInput 컴포넌트 생성

yonghyeun committed:
feat[#18] : OutLineTextInput storybook 오타 수정

yonghyeun committed:
feat[#18] : TImerTextInput 컴포넌트 생성

yonghyeun committed:
feat[#18] : OutLinedTextInput border-none 스타일 추가

yonghyeun committed:
feat[#18] : TimerTextInput storybook 파일 위치 수정

yonghyeun committed:
feat[#18] : OutLinedTextInput.stories 에서 아이콘 유무를 컨트롤 하기 위한 args 추가

yonghyeun committed:
feat[#18] : CalanderInput 컴포넌트 추가

- 달력 선택 시 생성 될 모달창 기본 브라우저 모달 사용 할 것인지 디자이너와 상의하기

yonghyeun committed:
fix[#18] : TimeTextInput story 오타 수정

yonghyeun committed:
fix[#18] : InpuutOutterProps 에서 title은 conditional types 로 수정

- SearcthTextInput 에선 title props 가 존재하지 않는다.

yonghyeun committed:
refactor[#18] : 공통 컴포넌트에서 props 를 props 객체 하나로만 받게 리팩토링

yonghyeun committed:
fix[#18] : ComponentType -> DesignType 으로 이름 수정

- ComponenType 은 react 에 이미 존재하는 타입이기 때문에 혼돈이 생길 수 있어 이름 수정

yonghyeun committed:
feat[#18] : DeisgnSystem 타입 추가

- 기존 발생하던 타입 에러는 DeisgnType 에서 현재 사용하지 않는 키가 여전히 존재했기 때문에 발생하던 타입 에러였음

yonghyeun committed:
refactor[#18] : Input DesignSystem 타입 리팩토링

yonghyeun committed:
fix[#18] : CalenderInput 오타 수정

yonghyeun committed:
feat[#18] : 스토리북 일부 내용 수정

- BaseInputStoires 삭제
- OutLInedTextInput 스토리에서 argTypes 수정
- SearchTextInput 스토리에서 args value -> defaultValue 수정
- TimerTextInput 스토리에서 svg 카멜 케이스로 수정

yonghyeun committed:
fix[#18] : TextInput placeholder props 추가

yonghyeun committed:
refactoring[#18] : plainTextInput -. TextInput 으로 이름 변경

yonghyeun committed:
feat[#18] : CalenderInput , SEarchTextInput 수정

- CalenderInput 스토리북 오타 수정
- SearchTextInput 스토리에서 withValue 스토리 삭제

yonghyeun committed:
feat[#18] : TextField 컴포넌트 추가

yonghyeun committed:
fix[#18] : TextInput 스토리에서 PlainTextInput -> TextInput 컴포넌트로 네이밍 수정

yonghyeun committed:
refactor[#18] : TextInput props 리팩토링

- 디스트럭처링이 아닌 props 객체 하나로 받도록 리팩토링

yonghyeun committed:
refactor[#18] : 디자인 시스템 컴포넌트 Props 넘겨주는 방식 변경

- props로 designType이 엉뚱한 값이 상위에서 넘어왔을 때 발생 할 문제를 방지하기 위해 디스트럭처링 시 기본값으로 값을 설정하는 것이 아니라 designType을 문자열로 BaesInput 에게 넘겨줌
- 스프레드 문법으로 남은 props를 넘겨주는 순서 변경,  props로 엉뚱한 값을 받아도 생성 시 의도한 props가 덮어씌워지도록 변경

yonghyeun committed:
fix[#18] : isSupportingNeeded boolean type 제거, supportingText 만으로 렌더링 하도록 수정

yonghyeun committed:
feat[#18] : 스토리북 Input.stories.tsx 파일 하나로 변경

- 코드 리뷰 시 도움을 위해 주석으로 설명 추가
- index.ts 를 통해 공용 API를 이용하여 외부로 export 하도록 변경

yonghyeun committed:
refactor[#18] : 대규모 리팩토링 , BaseInput 기반 컴포넌트 생성 로직 제거

- 추상화된 Input 컴포넌트 하나로 모든 Input 컴포넌트를 생성 가능하도록 수정
- 복잡하고 어려운 Input 타입 코드 제거
- props 목록 축소
- 의미있고 이해하기 쉬운 props로 네이밍 변경

yonghyeun committed:
fix[#18] : 스토리북 statusText 넣지 못했던 부분 수정

aken-you commented on src/shared/ui/input/Input.tsx:
```diff
@@ -0,0 +1,70 @@
+  label: string;
+  componentType: keyof typeof inputStyles;
+  isError: boolean;
+  statusText: null | string;
```
supportingText에서 statusText로 네이밍이 바뀌었네요!

yonghyeun replied:
네에엡 예전엔 디자인 시스템에 있는 문구를 그대로 작성했는데 해당 영역이 거의 유효성 검사나 API 요청 결과에 따른 문구를 나타낼거라서 이름을 바꿔보았습니닷

aken-you commented on src/shared/ui/input/Input.tsx:
```diff
@@ -0,0 +1,70 @@
+  leadingNode,
+  ...props
+}: InputProps) => {
+  // 휴먼에러를 방지하기 위해 leadingNode,trailign,leadingNode의 유효성 확인
```
꼼꼼한 예외처리까지 하셨군요bb
(주석에서 leadingNode가 하나 더 있어서 지워주시면 될 것 같습니다!)

aken-you commented on src/shared/ui/input/Input.tsx:
```diff
@@ -0,0 +1,70 @@
+  // input 컴포넌트의 상태에 따라 적용되는 스타일
+  const status: StatusType = isError ? "error" : "default";
+  const statusStyles = inputStyles[componentType][status];
+  const { statusTextclass, ...wrapperClasses } = statusStyles;
```
`statusTextClass` 오타난 것 같습니다!

aken-you commented on src/shared/ui/input/TextField.stories.tsx:
```diff
@@ -0,0 +1,30 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import TextField from "./TextField";
```
TextField 파일이 없어서 확인을 못했습니다ㅠ

yonghyeun replied:
추가 완료했습니다 ~

aken-you commented on src/shared/ui/input/input.styles.ts:
```diff
@@ -0,0 +1,141 @@
+      hover: "hover:border-grey-700 hover: bg-grey-100",
+      focus: "focus-within:border-tangerine-500 focus-within:bg-grey-0",
+      focusHover:
+        "hover:focus-within:border-tangerine-500 hover:focus-within:bg-grey-0",
```
focusHover가 어떤 상황일 때 적용되는 스타일인지 다시 한 번 설명해주실 수 있나요??

yonghyeun replied:
focusHover 클래스는 hover 이벤트와 focus 이벤트가 동시에 발생 했을 때를 위해 존재합니다. 

hover , focus 이벤트가 동시에 일어났을 때 (마우스를 위에 올려둔 채로 해당 input에 focus가 된 경우) hover 이벤트가 focus 이벤트보다 우선되어 적용되는 문제가 발생하더라고요 

그래서 두 이벤트가 동시에 일어났을 때의 스타일을 적용해보았습니다.

더 좋은 방법이 있을까요 ? 👁️👁️

aken-you commented on src/shared/ui/input/input.styles.ts:
```diff
@@ -0,0 +1,141 @@
+      statusTextclass: "",
+    },
+  },
+  outLinedTextInput: {
```
outLinedTextInput 대신 outlinedTextInput으로 수정해주시면 될 것 같습니다 :)

aken-you commented on src/shared/ui/input/Input.tsx:
```diff
@@ -0,0 +1,70 @@
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  id: string;
+  label: string;
+  componentType: keyof typeof inputStyles;
```
componentType 값들 중에서 끝에 Input이 있는 값이 있더라구요!
componentType은 Input의 prop이니까, 끝에 Input을 붙이지 않더라도 알 수 있을 거 같은데 이 부분은 어떻게 생각하시나요?!

ex) `searchTextInput` -> `searchText`

yonghyeun replied:
와우 아주 좋은 생각인 거 같습니다

yonghyeun replied:
모든 props 이름을 `input` 을 제거한 `"textField" | "calander" | "searchText" | "outlinedText" | "text" | "timerText"` 로 변경했습니다 ~~

aken-you commented on src/shared/ui/input/Input.tsx:
```diff
@@ -0,0 +1,70 @@
+  trailingNode,
+  leadingNode,
+  ...props
+}: InputProps) => {
```
저는 Button 컴포넌트의 props 타입을 Props로 썼었어요! 
앞으로 용현님처럼 props 타입 네이밍을 `컴포넌트이름Props`로 할까요?!

yonghyeun replied:
저는 `컴포넌트 이름 + Props` 로 하는게 좋을 거 같습니다 !! 이유는 명확해보이기도 하고 만약 타입을 외부로 export 해서 사용 할 때 더 명확해 질 수 있을 거 같습니닷

aken-you commented on src/shared/ui/input/Input.stories.tsx:
```diff
@@ -0,0 +1,452 @@
+        "input 엘리먼트 우측에 존재 할 수 있는 ReactNode입니다. trailingNode는 최대 두개의 노드만 가질 수 있습니다.",
+    },
+  },
+};
```
상세한 설명 덕분에 이해하기 쉬웠습니다 👍

yonghyeun replied:
친절한 코드리뷰깔쥐 .,., 코드리뷰 하는 법도 배워갑니다 줍줍

aken-you reviewed (commented):
storybook에 자세하게 설명해주셔서, 코드 이해하기 쉬웠습니다 👍
깔끔하게 리팩토링 해주셔서 읽기 편했어요!

근데 Text Area Field 관련한 컴포넌트가 보이지 않네요 👀 
추가해주시면 다시 리뷰 작성하겠습니다!

aken-you commented on src/shared/ui/input/input.styles.ts:
```diff
@@ -0,0 +1,141 @@
+      hover: "hover:border-pink-700 hover:bg-grey-100",
+      focus: "focus-within:border-pink-500 focus-within:bg-grey-0",
+      focusHover:
+        "hover:focus-within:border-pink-500 hover:foc us-within:bg-grey-0",
```
여기 fo cus로 오타났어요 👀

yonghyeun replied:
오마이갓 이제야 확인했네요 수정했습니다.

aken-you replied:
여기 아직 수정이 안된 것 같아요 👁️ 👁️

yonghyeun committed:
feat[#18] : 1차 코드 리뷰 사항에 따라 기능 변경

- statusTextClass 오타 수정
- 오타가 있던 주석 처리 수정
- Input 컴포넌트에서 존재하던 textField 부분 ui/TextArea 로 수정
- TextArea 컴포넌트 , 스토리북 생성

yonghyeun committed:
feat[18] : TextArea 컴포넌트 scrollbar 삭제

- 스크롤바는 추후 디자인이 추가 된 이후 커스텀 스타일로 적용 예정

yonghyeun committed:
fix[#18] : Input.styles.tsx fo cus  -> focus 오타 수정

yonghyeun committed:
fix[#18] : disable:plcaholder -> disabled:placehloder 수정

- input 태그 내에서 disabled 일 때 placeholder 색상이 변경되도록 오타 수정

yonghyeun committed:
feat[#18] : Input 컴포넌트 min-width-[20.5rem (328px)] 추가

- 기존 Input 컴포넌트에선 min-width가 설정되어 있지 않아 leadingIcon , tralingIcon 여부에 따라 너비가 다른 문제가 있었습니다.
- Input 부분 스토리북에서 실제 사용 예시를 추가하고 보기 좋게 수정했습니다.

yonghyeun committed:
fix[#18] : InputStyles 에서 TextArea에 해당하는 스타일 삭제

aken-you commented on src/shared/ui/input/input.styles.ts:
```diff
@@ -0,0 +1,119 @@
+      statusTextClass: "text-pink-500",
+    },
+  },
+} as const;
```
storybook에서 확인했을 때 disabled 상태인 input에 hover하면 hover 스타일이 적용돼요!

disabled이면 hover 해도 스타일이 변경되지 않도록 처리해야 할 것 같습니다

yonghyeun replied:
```tsx
const Input = ({
  id,
  label,
  componentType,
  isError = false,
  statusText = null,
  trailingNode,
  leadingNode,
  disabled = false,
  ...props
}: InputProps) => {
  // 휴먼에러를 방지하기 위해 leadingNode,trailign 유효성 확인
  if (React.Children.count(leadingNode) > 1) {
    throw new Error("leadingNode 하나의 노드만 가질 수 있습니다.");
  }
  if (React.Children.count(trailingNode) > 2) {
    throw new Error("trailingNode는 최대 2개의 노드만 가질 수 있습니다.");
  }

  // input 컴포넌트의 상태에 따라 적용되는 스타일
  const status: StatusType = isError ? "error" : "default";
  const statusStyles = inputStyles[componentType][status];
  const { statusTextClass, ...wrapperClassesObject } = statusStyles;

  // wrapper 에 해당하는 div에선 hover,focus에 대한 가상 선택자 스타일이 적용되지만
  // disabled에 대한 가상 선택자 스타일은 적용되지 않습니다.
  // 그렇기 때문에 disabled에 대한 스타일을 따로 분리하여 적용합니다.
  const { disabled: disableClass, ...restStylesObject } = wrapperClassesObject;
  const restClasess = Object.values(restStylesObject).join(" ");
  const wrapperClasses = [restClasess, disabled ? disableClass : ""].join(" ");
...
```

다음과 같이 수정했습니다. 

기존 코드에선 Input 태그에만 단순히 disabled 를 내려주었으나 wrapper 에 해당하는 div 태그엔 해당 가상 선택자가 작동하지 않아서 문제가 발생했던 것 같습니다. 

```tsx
      disabled: "bg-grey-0 border-grey-100 pointer-events-none",
```

모든 `InputStyle` 의 `disabled` 에서 가상선택자를 제거하고 `point-event-none` 클래스를 추가했습니다.

aken-you commented on src/shared/ui/textarea/TextArea.tsx:
```diff
@@ -0,0 +1,62 @@
+  maxLength = 150,
+  ...props
+}: TextAreaProps) => {
+  const [value, setValue] = useState<string>("");
```
TextArea을 사용하는 부모 컴포넌트에서 value 상태가 있을텐데, 여기서 선언하신 이유가 있나요?

yonghyeun replied:
처음에는 하단에 존재하는 글자수를 나타내는 곳에서 사용하기 위해 value , onChange를 부착했었습니다.

하지만 리뷰를 보고 나서 생각해보니 상단에서 value , onChange 를 주입 받은 경우 내부에 존재하는 이벤트 핸들러와 value 를 덮어씌워 내부 state 가 작동하지 않을 가능성이 있음을 판단하고 다음과 같이 변경했습니다.

```tsx
interface TextAreaProps
  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {
  id: string;
  label: string;
  isError: boolean;
  maxLength: number;
}

const TextArea = ({
  id,
  label,
  isError,
  maxLength = 150,
  ...props
}: TextAreaProps) => {
  const textAreaRef = useRef<HTMLTextAreaElement>(null);
  const [currentLength, setCurrentLength] = useState<number>(0);

  // textarea의 최대 글자수를 제한하고 현재 글자수를 표시하기 위한 Effect
  // 직접 actual dom에 접근하여 addEventListener를 사용하는 이유는 컴포넌트 상위에서 props로
  // 이벤트 핸들러를 등록해주는 경우가 존재 할 수 있기 때문입니다.
  // addEventListener로 이벤트 핸들러를 부착해두게 되면 상단에서 등록한 이벤트 핸들러와 충돌이 발생하지 않습니다.
  useEffect(() => {
    const $textArea = textAreaRef.current;
    if (!$textArea) {
      return;
    }

    // removeEventListener를 위한 이벤트 핸들러 함수로 기명 함수를 사용하여 removeEventListener를 사용할 수 있도록 합니다.
    const handleInput = (e: Event) => {
      const target = e.target as HTMLTextAreaElement;
      setCurrentLength(target.value.length);
    };

    $textArea.addEventListener("input", handleInput);

    return () => {
      $textArea.removeEventListener("input", handleInput);
    };
  }, [maxLength]);

  const status = isError ? "error" : "default";
  const wrapperClasses = textAreaStyles[status];

  return (
    <div className="flex flex-col items-start">
      <label htmlFor={id} className="title-3 flex items-start gap-1 pb-2">
        {label}
      </label>
      <div className={`${Object.values(wrapperClasses).join(" ")}`}>
        <textarea
          name={id}
          id={id}
          className={baseStyles.textArea}
          maxLength={maxLength}
          ref={textAreaRef}
          {...props}
        />
        <p className="flex items-end justify-end gap-[2px] self-stretch">
          {`${currentLength}/${maxLength}`}
        </p>
      </div>
    </div>
  );
};

export default TextArea;
```

`addEventListener , removeEventListener` 를 이용하여 이벤트 핸들러가 상단에서 주입되더라도 중첩 가능하게 하였고 

상태의 존재 이유를 더 명확히 하기 위해 상태를 `currentLength` 로 변경하여 사용했습니다.

aken-you replied:
부모 컴포넌트로부터 value를 받으면 currentLength 값을 value.length로 해도 될 것 같은데, 내부에 currentLength라는 상태를 두신 이유가 있나요?!
```javascript
function Component() {
  const [value, setValue] = React.useState("");
  return (
    <div>
      <TextArea
        value={value}
        onChange={(e) => setValue(e.target.value)}
        maxLen={150}
      />
    </div>
  );
}

function TextArea({ value, onChange, maxLen }) {
  return (
    <div>
      <textarea value={value} onChange={onChange} maxLength={maxLen}></textarea>
      <p>
        {value.length}/{maxLen}
      </p>
    </div>
  );
}



```

aken-you commented on src/shared/ui/textarea/TextArea.tsx:
```diff
@@ -0,0 +1,62 @@
+          maxLength={maxLength}
+          style={{
+            backgroundColor: "transparent",
+          }}
```
style 객체를 사용하신 이유가 있을까요?!

yonghyeun replied:
tailwind 에서 input 태그의 background 를 transparent , inherit 로 변경하는 것이 불가능하더라고요 

```css
input,
textarea {
  background-color: inherit;
}
```

다만 컴포넌트 내에서 인라인 스타일을 작성하는 것이 컴포넌트의 가독성을 떨어뜨려 해당 부분을 제거하고 global.css 에서 설정해놓는 것으로 변경하겠습니다.

yonghyeun replied:
`global.css` 에 정의하도록 변경했습니다.

aken-you commented on src/shared/ui/textarea/TextArea.stories.tsx:
```diff
@@ -0,0 +1,59 @@
+        "TextArea 컴포넌트에서 입력 가능한 최대 글자수를 나타냅니다. 기본 값은 150입니다.",
+    },
+  },
+};
```
argTypes에 disabled도 추가해주시면 좋을 것 같아요!
disabled일 때, 스타일을 확인하고 싶어서요 👀

aken-you reviewed (commented):
textarea까지 잘 봤습니다! 이번에도 확인 부탁드릴게요~

yonghyeun committed:
fix[#18] : Input min-width 제거

- flex box로 input 태그를 감싸는 컨테이너를 모두 만족하도록 수정

yonghyeun committed:
fix[#18] : wrapper class items-center -> start 수정

yonghyeun committed:
refactor[#18] : Input ,TextArea 배경색 인라인 스타일 제거

yonghyeun committed:
feat[#18] : textArea 이벤트 핸들러 중첩을 방지하기 위해 useEffect 사용

- textarea 의 글자수를 세기 위한 state 를 currentLength로 의미론적으로 부합한 상태로 변경

yonghyeun committed:
feat[#18] : Input calender 기능 추가

- 기본 indicator calender 이미지 assets/calender.svg 파일로 배경 변경 및 css 설정
- storybook componentType props 변경 (calender -> date)

yonghyeun committed:
feat[#18] : Input 컴포넌트 disabled props 추가

- div 태그와 같이 active 하지 않은 태그는 disabled 가상 선택자 스타일 속성이 적용되지 않음
- 그렇기 때문에 disabled 와 같은 이벤트를 흉내내는 클래스 속성 추가 및 핸들링

aken-you commented on chromatic.config.json:
```diff
@@ -0,0 +1,5 @@
+  "onlyChanged": true,
+  "projectId": "Project:66bae48876a5167719ee1778",
+  "zip": true
+}
```
이건 어떤 설정인가요 ?

aken-you commented on src/global.css:
```diff
@@ -51,3 +51,36 @@ body {
+/* Edge, Internet Explorer */
+textarea {
+  -ms-overflow-style: none; /* IE and Edge */
+}
```
textarea에서 안의 내용이 길어질 경우 나타나는 스크롤 스타일을 없애신거 맞나요?!

yonghyeun replied:
맞습니다

aken-you commented on src/shared/ui/textarea/TextArea.tsx:
```diff
@@ -0,0 +1,73 @@
+
+    // removeEventListener를 위한 이벤트 핸들러 함수로 기명 함수를 사용하여 removeEventListener를 사용할 수 있도록 합니다.
+    const handleInput = (e: Event) => {
+      const target = e.target as HTMLTextAreaElement;
```
as로 타입을 정의한 이유가 있나요?!

yonghyeun replied:
그냥 타입가드 용으로 했습니다.

aken-you replied:
```javascript
const handleInput = (e: Event) => {
      const { value } = e.target as HTMLTextAreaElement;
      setCurrentLength(value.length);
};
```

이렇게도 쓸 수 있을 거 같네요!

yonghyeun replied:
해당 부분 맞춰 리팩토링 진행했습니다 ~!!

aken-you commented on src/shared/ui/textarea/TextArea.tsx:
```diff
@@ -0,0 +1,73 @@
+    return () => {
+      $textArea.removeEventListener("input", handleInput);
+    };
+  }, [maxLength]);
```
textAreaRef를 callback 함수에서 사용하는데 의존성 배열에 안넣은 이유가 있나요?!

yonghyeun replied:
어차피 ref 는 불변하니까 안넣어도 될 거 같았습니다.

yonghyeun committed:
fix[#18] : Input story 내부 Calender 컴포넌트 제거

aken-you commented on src/shared/ui/textarea/TextArea.tsx:
```diff
@@ -0,0 +1,73 @@
+    $textArea.addEventListener("input", handleInput);
+
+    return () => {
+      $textArea.removeEventListener("input", handleInput);
```
input이랑 change 이벤트 차이 알아왔습니다
([참고](https://ko.javascript.info/events-change-input))

- change: 텍스트 입력 요소인 경우, 값을 입력하는 동안 이벤트가 발생하지 x. 입력을 다하고 나서 outfocus하면 이벤트 발생

- input: 사용자가 값을 수정할 때마다 이벤트 발생

yonghyeun replied:
감사합니다

aken-you replied:
다시 찾아보니 javascript에서만 input이랑 change 이벤트 차이가 있는거였네요.

react에서는 컴포넌트의 onChange나 onInput의 큰 차이가 없다고 합니다.

yonghyeun replied:
확인했습니다 !!! 좋은 정보 감사합니다

yonghyeun committed:
feat[#18] : TextArea Disabled 기능 추가

aken-you commented on src/shared/ui/textarea/TextArea.tsx:
```diff
@@ -0,0 +1,73 @@
+  isError,
+  maxLength = 150,
+  ...props
+}: TextAreaProps) => {
```
부모 컴포넌트에서 참조할 수 있어서 forwardRef 씌워줘야 할 것 같습니다 :)

yonghyeun replied:
`forwardRef` 하려면 대규모 수정 작업이 일어나야 해서 추후 `ref` 가 필요한 순간 해보면 어떨까요 ?

aken-you commented on src/shared/ui/input/Input.tsx:
```diff
@@ -0,0 +1,77 @@
+  disabled: boolean;
+}
+
+const Input = ({
```
여기도 forwardRef로 씌워주면 좋을 것 같아요!

yonghyeun replied:
`forwardRef` 하려면 대규모 수정 작업이 일어나야 해서 추후 `ref` 가 필요한 순간 해보면 어떨까요 ?

yonghyeun committed:
fix[#18] : TextArea 오타 수정

yonghyeun committed:
fix[#18] : TextArea enabled class 수정

yonghyeun committed:
feat[#18] : disabled , enabled에 따른 class 명 분기 로직 수정

- 현재 tailwindcss 에선 css 가 덮어씌워질 때 토큰 숫자가 높은 것으로 덮어씌워짐
- 이를 해결하기 위해 enabled , disabled 일 때에 따라 스타일 분기 처리

aken-you commented:
Input이나 TextArea를 내보내는 index.ts 파일은 필요할까요?

저는 우선 ui/button/index.tsx에서 Button 컴포넌트 내보냈어요 (index.ts에서 내보냈어야 했나 싶네요 💦 )

yonghyeun committed:
refactor[#18] : Input , TextArea 리팩토링

- 불필요한 변수 선언 제거
- 디스트럭처링 활용하여 코드 라인 줄이기

yonghyeun commented:
> Input이나 TextArea를 내보내는 index.ts 파일은 필요할까요?
> 
> 저는 우선 ui/button/index.tsx에서 Button 컴포넌트 내보냈어요 (index.ts에서 내보냈어야 했나 싶네요 💦 )

앗 그르게요 저는 `index.ts` 파일을 만들어두지 않았는데 `index.ts` 파일에서 컴포넌트 작업을 해야겠습니다 

저도 `index.ts` 에서 `export` 하도록 하겠습니다

yonghyeun committed:
fix[#18] : shared 레이어 index.tsx 공용 API 파일을 통해 export 하도록 수정

yonghyeun committed:
fix[#18] : index.ts 공용 API 를 이용한 export

yonghyeun committed:
fix[#18] : storybook import 경로 수정
